mlir-tensorrt/tensorrt/lib/TensorRT/IR/TypeInferenceInterfaceImpls.cpp

```diff
@@ -123,6 +123,27 @@ LogicalResult tensorrt::ElementWiseOp::inferReturnTypeComponents(
   return success();
 }
 
+
+//===----------------------------------------------------------------------===//
+// CastOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult tensorrt::CastOp::inferReturnTypeComponents(
+  MLIRContext *ctx, std::optional<Location> loc, ValueShapeRange operands,
+  DictionaryAttr attributes, OpaqueProperties properties, RegionRange regions,
+  SmallVectorImpl<ShapedTypeComponents> &inferredReturnShapes) {
+// TODO (pranavm): Check if we need to do something here. Identity does not seem
+// to infer the output element type.
```

Comment:
Not entirely sure what to do here. This seems to only be inferring the shape for `identity`.